File: tools/improve/tests/original/maudec/py/even.py


def even1(N: int) -> bool:
	return N % 2 == 0


def even2(a1: int) -> bool:
	match a1:
		case 0:
			return True

		case 1:
			return False

		case N if N >= 2:
			return even2(N - 2)

File: tools/improve/tests/original/maudec/cpp/even.cpp



bool even1(int N) {
	return N % 2 == 0;
}

bool even2(int a1) {
	if (a1 == 0) {
		return true;
	}
	if (a1 == 1) {
		return false;
	}
	if (a1 >= 2) {
		return even2(a1 - 2);
	}
}

